spec/measure_repository_service_test_kit/read_by_id_spec.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+RSpec.describe MeasureRepositoryServiceTestKit::ReadById do
+  let(:suite) { Inferno::Repositories::TestSuites.new.find('measure_repository_service_test_suite') }
+  let(:group) { suite.groups[1] }
+  let(:session_data_repo) { Inferno::Repositories::SessionData.new }
+  let(:test_session) { repo_create(:test_session, test_suite_id: suite.id) }
+  let(:url) { 'http://example.com/fhir' }
+  let(:error_outcome) { FHIR::OperationOutcome.new(issue: [{ severity: 'error' }]) }
+
+  def run(runnable, inputs = {})
+    test_run_params = { test_session_id: test_session.id }.merge(runnable.reference_hash)
+    test_run = Inferno::Repositories::TestRuns.new.create(test_run_params)
+    inputs.each do |name, value|
+      session_data_repo.save(test_session_id: test_session.id, name:, value:, type: 'text')
+    end
+    Inferno::TestRunner.new(test_session:, test_run:).run(runnable)
+  end
+
+  describe 'Server successfully retrieves specified resource by its id' do
+    let(:test) { group.tests.first }
+    let(:resource_id) { 'measure_id' }
+
+    it 'passes if a Measure resource with the specified id was received' do
+      resource_type = 'Measure'
+      resource = FHIR::Measure.new(id: resource_id)
+      stub_request(
+        :get,
+        "#{url}/#{resource_type}/#{resource_id}"
+      ).to_return(status: 200, body: resource.to_json)
+
+      result = run(test, url:, resource_type:, resource_id:)
+      expect(result.result).to eq('pass')
+    end
+
+    it 'passes if a Library resource with the specified id was received' do
+      resource_type = 'Library'
+      resource = FHIR::Library.new(id: resource_id)
+      stub_request(
+        :get,
+        "#{url}/#{resource_type}/#{resource_id}"
+      ).to_return(status: 200, body: resource.to_json)
+
+      result = run(test, url:, resource_type:, resource_id:)
+      expect(result.result).to eq('pass')
+    end
+
+    it 'fails if read interaction does not return 200' do
+      resource_type = 'Measure'
+      resource = FHIR::Measure.new(id: resource_id)
+      stub_request(
+        :get,
+        "#{url}/#{resource_type}/#{resource_id}"
+      ).to_return(status: 400, body: resource.to_json)
+      result = run(test, url:, resource_type:, resource_id:)
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if the id received does not match the one requested' do
+      resource_type = 'Measure'
+      resource = FHIR::Measure.new(id: 'INVALID_ID')
+      stub_request(
+        :get,
+        "#{url}/#{resource_type}/#{resource_id}"
+      ).to_return(status: 200, body: resource.to_json)
+      result = run(test, url:, resource_type:, resource_id:)
+      expect(result.result).to eq('fail')
+    end
+  end
+
```

Comment:
Can we add a test for expected failure when the returned resource doesn't match the the passed-in resource type